### PR TITLE
Params

### PR DESCRIPTION
**Pull request by lodr**

### CODE REVIEW

**Comment by lodr** [comment]
@arcturus do you mind to take a look?

File: lib/router.js [hunk 1]
```diff
@@ -62,13 +65,15 @@ Router.prototype.match = function r_match(method, url) {
...
   this.stack.forEach(function eachRoute(route) {
-    if (!(method === route.method || route.method === self.ALL_METHODS)) {
+    if (!(method === route.method || route.method === _this.ALL_METHODS)) {
       return;
     }
 
-    if (route.path.test(url)) {
+    var groups = _this._routeMatch(url, route);
+    if (groups) {
+      route.handler.__params = groups;
```
Comment (arcturus) [inline lib/router.js]: The handler can be a function too, not just a middleware object, are we safe for that option too?

Comment (lodr) [inline lib/router.js]: You can add properties to functions for free. :)

**Comment by lodr** [comment]
@arcturus do you think we can merge? Or do you miss something?

**Comment by arcturus** [comment]
Yes I do, not a great fan of adding properties to functions }:) but we definitely can merge

**Pull request merged**